core/crypto/vrf/p256/p256_test.go

```diff
@@ -66,6 +74,60 @@ func TestH2(t *testing.T) {
 	}
 }
 
+func TestFromWrappedKey(t *testing.T) {
+	ctx := context.Background()
+	for _, tc := range []struct {
+		spec   *keyspb.Specification
+		keygen keys.ProtoGenerator
+	}{
+		{
+			spec: &keyspb.Specification{
+				Params: &keyspb.Specification_EcdsaParams{
+					EcdsaParams: &keyspb.Specification_ECDSA{
+						Curve: keyspb.Specification_ECDSA_P256,
+					},
+				},
+			},
+			keygen: func(ctx context.Context, spec *keyspb.Specification) (proto.Message, error) {
+				return der.NewProtoFromSpec(spec)
+			},
+		},
+	} {
+		// Generate VRF key.
+		wrapped, err := tc.keygen(ctx, tc.spec)
+		if err != nil {
+			t.Errorf("NewProtoFromSpec failed: %v", err)
+			continue
+		}
+		vrfPriv, err := NewFromWrappedKey(ctx, wrapped)
+		if err != nil {
+			t.Errorf("NewFromWrappedKey failed: %v", err)
+			continue
+		}
+		vrfPublicPB, err := der.ToPublicProto(vrfPriv.Public())
+		if err != nil {
+			t.Errorf("ToPublicProto failed: %v", err)
+			continue
+		}
+		vrfPub, err := NewVRFVerifierFromRawKey(vrfPublicPB.Der)
+		if err != nil {
+			t.Errorf("NewVRFVerifierFromRawKey failed: %v", err)
+			continue
+		}
+		// Test that the public and private components match.
+		m1 := []byte("foobar")
+		index1, proof1 := vrfPriv.Evaluate(m1)
```

Comment:
s/index1/privIndex/
s/proof1/proof/

Reply:
done

Reply:
Index isn't private, so I've labeled these `indexA` and `IndexB`